### ISSUE TITLE
Chained requests works in desktop client but not via Inso CLI

### ISSUE THREAD

**Issue by 0erty** [getinsomnia/insomnia#7453]
### Expected Behavior

Chaining requests works in CLI as well as on desktop

### Actual Behavior

The same as here https://github.com/Kong/insomnia/issues/3366

I have checked the same chaining requests on inso cli v.2.4.0 and 9.2.0 and the results are different. Everything works on v.2.4.0 and the same requests don't work on v.9.2.0.
**v.2.4.0:**
<img width="1134" alt="image" src="https://github.com/Kong/insomnia/assets/32879454/72bd834a-5f85-476e-83b1-156a045e7c3e">
**v.9.2.0:**
<img width="566" alt="image" src="https://github.com/Kong/insomnia/assets/32879454/5594a803-35e9-4e2f-a537-46759d8f7296">

### Reproduction Steps

1. Make a request that returns something
2. Make a second request that needs something from the first
3. Make a test suite that contains those two requests, each as "tests"
4. Verify Run Tests works
5. Try and run inso test and select that suite

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Additional Information

- Installation Method: Mac App

```
Version: Insomnia 9.2.0
Build date: 5/15/2024
OS: Darwin arm64 23.4.0
Electron: 29.3.0
Node: 20.9.0
Node ABI: 121
V8: 12.2.281.27-electron.0
Architecture: arm64
```

### Insomnia Version

9.2.0 / Inso CLI 9.2.0

### What operating system are you using?

macOS

### Operating System Version

mac OS Sonoma 14.4.1 

### Installation method

Inso CLI

### Last Known Working Insomnia version

3.8.0

**Comment by gatzjames** [comment]
Hi @0erty,
Are you using the `Response Body` tag from another request or do you store the value in the environment?
The second option is not supported ([related Issue](https://github.com/Kong/insomnia/issues/6892))
Otherwise you might want to check if the `Trigger behavior` in the `Response Body` tag's settings is not set to `Never` as this could run fine in Insomnia because you have a previous response but not on the cli

**Comment by 0erty** [comment]
Hi @gatzjames, I tried it in both options and it doesn't work. We can see that it works in desktop but not in CLI. And I use "Always" setting in Trigger behaviour. 
<img width="1100" alt="image" src="https://github.com/Kong/insomnia/assets/32879454/323d928c-70ab-4e39-88ab-a2546f583dc6">

**Comment by gatzjames** [comment]
Could you run the CLI with the --verbose flag?
This will provide more context into what is happening

**Comment by 0erty** [comment]
@gatzjames fyi
```
Collection
[plugin] Loading 
[response tag] Resending dependency
Error rendering template Template render error: (unknown path) [Line 1, Column 1]
  attempted to output null or undefined value
    at Object._prettifyError (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
    at /snapshot/insomnia/packages/insomnia-send-request/dist/index.js
    at Template2.root [as rootRenderFunc] (eval at _compile (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js), <anonymous>:18:3)
    at Template2.render3 [as render] (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
    at Environment4.renderString (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
    at /snapshot/insomnia/packages/insomnia-send-request/dist/index.js
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
Failed to render element url {{ _.base_url }}/client-api/v1/auth/login
Error rendering template Template render error: (unknown path)
  Error: attempted to output null or undefined value
    at Object._prettifyError (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
    at /snapshot/insomnia/packages/insomnia-send-request/dist/index.js
    at eval (eval at _compile (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js), <anonymous>:11:11)
    at /snapshot/insomnia/packages/insomnia-send-request/dist/index.js
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
Failed to render element Environment cookie1={% response 'body', 'req_9a7e36ca9dc54432b5253d61fad75936', 'b64::JC5tb25vbGl0aC5jb29raWU=::46b', 'always', 60 %}; Expires=Tue, 19 Jan 2038 03:14:07 GMT; Domain=appfollow.io; Path=/
[response tag] Resending dependency
Error rendering template Template render error: (unknown path) [Line 1, Column 1]
  attempted to output null or undefined value
    at Object._prettifyError (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
    at /snapshot/insomnia/packages/insomnia-send-request/dist/index.js
    at Template2.root [as rootRenderFunc] (eval at _compile (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js), <anonymous>:18:3)
    at Template2.render3 [as render] (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
    at Environment4.renderString (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
    at /snapshot/insomnia/packages/insomnia-send-request/dist/index.js
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
Failed to render element url {{ _.base_url }}/client-api/v1/auth/login
Error rendering template Template render error: (unknown path)
  Error: attempted to output null or undefined value
    at Object._prettifyError (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
    at /snapshot/insomnia/packages/insomnia-send-request/dist/index.js
    at eval (eval at _compile (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js), <anonymous>:11:11)
    at /snapshot/insomnia/packages/insomnia-send-request/dist/index.js
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
Failed to render element Environment cookie2={% response 'body', 'req_9a7e36ca9dc54432b5253d61fad75936', 'b64::JC5jbGllbnRfYXBpLmFjY2Vzc190b2tlbg==::46b', 'always', 60 %}; Expires=Tue, 19 Jan 2038 03:14:07 GMT; Domain=appfollow.io; Path=/
[response tag] Resending dependency
Error rendering template Template render error: (unknown path) [Line 1, Column 1]
  attempted to output null or undefined value
    at Object._prettifyError (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
    at /snapshot/insomnia/packages/insomnia-send-request/dist/index.js
    at Template2.root [as rootRenderFunc] (eval at _compile (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js), <anonymous>:18:3)
    at Template2.render3 [as render] (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
    at Environment4.renderString (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
    at /snapshot/insomnia/packages/insomnia-send-request/dist/index.js
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
Failed to render element url {{ _.base_url }}/client-api/v1/auth/login
Error rendering template Template render error: (unknown path)
  Error: attempted to output null or undefined value
    at Object._prettifyError (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
    at /snapshot/insomnia/packages/insomnia-send-request/dist/index.js
    at eval (eval at _compile (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js), <anonymous>:11:11)
    at /snapshot/insomnia/packages/insomnia-send-request/dist/index.js
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
Failed to render element headers[0].value cookie1={% response 'body', 'req_9a7e36ca9dc54432b5253d61fad75936', 'b64::JC5tb25vbGl0aC5jb29raWU=::46b', 'always', 60 %};cookie2={% response 'body', 'req_9a7e36ca9dc54432b5253d61fad75936', 'b64::JC5jbGllbnRfYXBpLmFjY2Vzc190b2tlbg==::46b', 'always', 60 %}
    1) Check collection


  0 passing (25ms)
  1 failing

  1) Collection
       Check collection:
     Error: attempted to output null or undefined value
      at /snapshot/insomnia/packages/insomnia-send-request/dist/index.js
      at /snapshot/insomnia/packages/insomnia-send-request/dist/index.js
      at RawTask.call (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
      at flush (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
      at Timeout.handleTimer (/snapshot/insomnia/packages/insomnia-send-request/dist/index.js)
      at listOnTimeout (node:internal/timers:573:17)
      at process.processTimers (node:internal/timers:514:7)
```

**Comment by gatzjames** [comment]
@0erty from the logs it looks like there might be a missing variable in the selected environment?

> [response tag] Resending dependency

Request chaining seems to work since inso is trying to send the request but fails to find an env variable?

**Comment by 0erty** [comment]
@gatzjames This variable is in the environment, but it seems that Insomnia does not write the received value from the previous request to it. I downgraded to version 3.8, everything works fine there

**Comment by gatzjames** [comment]
Yep storing a chained request result to the environment is not supported by inso since it would need to recursively evaluate the environment. In that version I think we did a double pass but that led to other substantial issues and was removed.
I'd suggest using the response tag directly in the places you want to use the response instead of the environment.
We are also working on pre-request/after-response scripts atm so once these are available in inso you will be able to manipulate the env from there.
